src/covid_model_seiir_pipeline/io/data_roots.py

```diff
@@ -0,0 +1,109 @@
+from pathlib import Path
+from typing import List, Union
+
+from .keys import (
+    DatasetType,
+    MetadataType,
+    DatasetKey,
+    LEAF_TEMPLATES,
+    PATH_TEMPLATES,
+)
+from .marshall import (
+    DATA_STRATEGIES,
+    METADATA_STRATEGIES,
+)
+
+
+class DataRoot:
+
+    def __init__(self, root: Union[str, Path], data_format: str = 'csv', metadata_format: str = 'yaml'):
+        self._root = Path(root)
+        if data_format not in DATA_STRATEGIES:
+            raise
+        self._data_format = data_format
+        if metadata_format not in METADATA_STRATEGIES:
+            raise
+        self._metadata_format = metadata_format
+
+    def dataset_types(self) -> List[str]:
+        return [dataset_name for dataset_name, attr in type(self).__dict__.items()
+                if isinstance(attr, DatasetType)]
+
+    def metadata_types(self) -> List[str]:
+        return [metadata_name for metadata_name, attr in type(self).__dict__.items()
+                if isinstance(attr, MetadataType)]
+
+
+class InfectionRoot(DataRoot):
+    metadata = MetadataType('metadata')
+
+    def modeled_locations(self) -> List[int]:
+        """Retrieve all of the location specific infection directories."""
+        return [int(p.name.split('_')[-1]) for p in self._root.iterdir() if p.is_dir()]
+
+    @property
+    def infections(self):
```

Comment:
hack hack hack

Reply:
If I'm reading this correctly you've implemented `infection_root_instance.infections(loc_id, draw_id)` in the weirdest way possible.Why this way?

Reply:
I was going to see if I could mimic the implementation of the Datatype where `data_root_instance.datatype` returns a callable.  But that's dumb.  Program to the interface is the right move.  I blame lateness.

Reply:
made this less dumb